### PR TITLE
Allow Roddy to skip full job manager init for most start modes

### PR DESCRIPTION
**Pull request by dankwart-de**
e.g. testrun/testrerun etc. do not need to init a cluster job manager.
In these cases, the direct sync job manager will be used.

### CODE REVIEW

File: RoddyCore/src/de/dkfz/roddy/Roddy.java [hunk 1]
```diff
@@ -276,17 +278,19 @@ private static void startup(String[] args) {
...
-            time("parsemode");
-            performCLIExit(clc.startupMode);
-        } else {
+        if (!initializeServices(clc.startupMode)) {
             logger.postAlwaysInfo("Could not initialize services. Roddy will not execute jobs.");
             performCLIExit(clc.startupMode, 1);
         }
+
+        if (!jobExecutionRequirementsFulfilled())
+            exit(1);
```
Comment (vinjana) [inline RoddyCore/src/de/dkfz/roddy/Roddy.java]: No message here?

File: RoddyCore/src/de/dkfz/roddy/client/RoddyStartupModeScopes.java [hunk 2]
```diff
@@ -19,16 +19,30 @@
...
      */
-    public static final int SCOPE_FULL = 0x20;
+    SCOPE_FULL(true),
 
+    /**
+     * This scope starts all modes and also initializes the Job Manager
+     */
+    SCOPE_FULL_WITHJOBMANAGER(true);
+
+    public final boolean needsFullInit;
```
Comment (vinjana) [inline RoddyCore/src/de/dkfz/roddy/client/RoddyStartupModeScopes.java]: I see a conceptual problem here. 'full' kind of implies that the JM is also initialized as requested. As we have it in the moment, different configuration toggles/alternatives blend in this multifunctional enum. So the current design fails to implement the single responsibility principle.

We could initialize each start-up mode with just a set of toggles that indicate, whether or not specific features should be turned on or of (or independent enums, if its not just toggles). Would that make sense?

File: RoddyCore/src/de/dkfz/roddy/Roddy.java (hunk 1 above)
Comment (dankwart-de) [reply RoddyCore/src/de/dkfz/roddy/Roddy.java]: Errors inside :D

File: RoddyCore/src/de/dkfz/roddy/client/RoddyStartupModeScopes.java (hunk 2 above)
Comment (dankwart-de) [reply RoddyCore/src/de/dkfz/roddy/client/RoddyStartupModeScopes.java]: As discussed, I'll leave it.

**Comment by dankwart-de** [comment]
Ok, as discussed some minutes ago, I'd leave it now and merge it.

**Comment by vinjana** [comment]
Okidoki.

**Pull request merged**